FFmpegInterop/SubtitleProvider.h

```diff
@@ -329,6 +337,7 @@ namespace FFmpegInterop
 
 			if (timer != nullptr)
 			{
+				timer->Tick -= tickToken;
```

Comment:
I think this is a bug: You detach the event handler here, but actually the timer should only be stopped. The event handler is never registered again. It must stay attached.

Just as a remark: I have recently read that EventHandlers in C++/CX (unlike C#) are realized as weak references. Microsoft docs actually recommend that you do not detach event handlers yourself for cleanup. Otherwise you are coding stuff yourself that the runtime would automatically do. Only exception is when you explicitly want to stop certain events, so your object lives on, but due to state you do not want certain events now. Here the timer is stopped, so it will not raise any events anyways, until it is started again.

Using WeakReference in the StartTimer method is correct! Here we have a "manual" delegate, which would create a strong reference.

Reply:
Hmm interesting find. Should we actually stop detaching all the event handlers around the library?

Reply:
It does not really hurt to leave them in, but we should not introduce new ones. I already removed a few during recent changes on subtitle classes.

Reply:
You are right that is a bug. I did the wrong merge there. I will delete it completely since it is not needed now that you found the thing about detaching in c++/cx.